fix: Restore empty files when healing

vadmeste opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
HealFile() does not process the case when an empty file is lost in
some disks. Since, Reedsolomon erasure doesn't handle restoring empty
data, HealFile() will create empty files similarly to CreateFile().


## Motivation and Context
Fixes #5069 

## How Has This Been Tested?
go test + Manually

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

aead commented on cmd/erasure-healfile.go:
```diff
@@ -117,13 +116,18 @@ func (s ErasureStorage) HealFile(staleDisks []StorageAPI, volume, path string, b
+			if err = s.ErasureDecodeDataAndParityBlocks(blocks); err != nil {
+				return f, err
+			}
+		} else {
```
I think this else is not required. If `chunksize == 0` than `len(blocks[i]) == 0.

vadmeste replied:
> I think this else is not required. If chunksize == 0 than `len(blocks[i]) == 0.

Yeah.. but sometimes blocks[i] is empty and sometimes it is nil because the previous loop tries to exit as fast as possible. My policy is to make things consistent in a such random behavior to avoid bugs..

aead replied:
I do not agree that it would be sometimes nil and sometimes empty. It should be always empty?! We never assign `nil` to a block, right?! However I think it's still fine for explicitness.

aead reviewed (approved):
LGTM

donatello reviewed (changes requested):
See comment https://github.com/minio/minio/pull/5257#issuecomment-350129004

vadmeste committed:
fix: Restore empty files when healing

HealFile() does not process the case when an empty file is lost in
some disks. Since, Reedsolomon erasure doesn't handle restoring empty
data, HealFile will create empty files similarly to CreateFile().

donatello reviewed (dismissed):
LGTM

donatello commented on cmd/erasure-healfile.go:
```diff
@@ -121,9 +123,14 @@ func (s ErasureStorage) HealFile(staleDisks []StorageAPI, volume, path string, b
+		// parity shards - but we skip this step if we are
+		// reconstructing an empty file.
+		if chunksize > 0 {
+			// reconstruct data - this computes all data and parity shards
```
We could remove this extra comment.

vadmeste replied:
> We could remove this extra comment.

Done

aead reviewed (approved):
LGTM
